pkg/sfu/receiver.go

```diff
@@ -638,14 +588,12 @@ func (w *WebRTCReceiver) storeDownTrack(track TrackSender) {
 	w.downTrackMu.Lock()
 	defer w.downTrackMu.Unlock()
 
-	for idx := range w.free {
-		w.index[track.PeerID()] = idx
+	idx, foundFree := w.downTrackSpreader.Store(track.PeerID(), len(w.downTracks))
```

Comment:
looks strange on my side. the index/free is managed by DownTrackSpreader but used in WebRTCReceiver, and Spreader rely on WebRTCReceiver's lock to protect the two fields. How about pass len(w.free) as argument to Spreader.Broadcast?

Reply:
Agreed. Have to repeat that code on relay side. Was trying to avoid it. Maybe downTracks can also be moved in there. What do you think?

Reply:
I thinks that's good, manage all these fields by Spreader or just extract parallel write function

Reply:
Cool, I will manage everything in there as we can re-use that code in relay up track.

Reply:
@cnderrauber Addressed in this commit - https://github.com/livekit/livekit-server/pull/657/commits/9dcd5f801b48641871fbfb2fe26bdf7d774499c5. Let me know what you think.